marlowe/src/Language/Marlowe/Extended/V1/Metadata/Types.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Language.Marlowe.Extended.V1.Metadata.Types
+  where
+
+import Data.Aeson (FromJSON, Value, parseJSON, withArray, withObject, (.:))
+import Data.Aeson.Types (Parser)
+import Data.Bifunctor (Bifunctor(first))
+import qualified Data.Foldable as F
+import Data.Map (Map)
+import qualified Data.Map as Map
+import Data.Map.Ordered (OMap)
+import qualified Data.Map.Ordered as OMap
+import Data.Set (Set)
+import Data.Set.Ordered (OSet)
+import Data.Text.Encoding as Text
+import qualified GHC.Generics
+import Plutus.V1.Ledger.Api (TokenName)
+import qualified Plutus.V1.Ledger.Value as Val
+
+data ContractType
```

Comment:
This isn't much of a taxonomy. Do we really need a restrictive `ContractType` like this?

Reply:
same comment as above